src/sentry/eventstream/kafka.py

```diff
@@ -113,13 +113,12 @@ def _send(self, project_id, _type, extra_data=(), asynchronous=True):
 
     def insert(self, group, event, is_new, is_sample, is_regression,
                is_new_group_environment, primary_hash, skip_consume=False):
-        # ensure the superclass's insert() is called, regardless of what happens
-        # attempting to send to Kafka
-        super(KafkaEventStream, self).insert(
-            group, event, is_new, is_sample,
-            is_regression, is_new_group_environment,
-            primary_hash, skip_consume
-        )
+        if options.get('eventstream.kafka.send-post_process-task'):
```

Comment:
Any reason not to move this check into `base.EventStream` instead? Same difference (for now?) but it was just one more level of indirection to understanding.

I'm probably missing something though.

Reply:
This could be in both, but I think it's more appropriate here. This super call was always a little weird and didn't exactly belong, because we needed the return value and couldn't get it if we were using this as a secondary backend (more context in GH-10078.) I can't think of a reason you'd want to use the base backend but /not/ enqueue the task?